src/box/tuple_constraint_fkey.c

```diff
@@ -179,21 +179,28 @@ key_info_by_field_no(const void *ptr1, const void *ptr2)
 }
 
 /**
- * Get of extract key for foreign index from local tuple by given as @a mp_data.
+ * Get or extract key for foreign index from local tuple by given as @a mp_data.
  * Simply return mp_data for field foreign key - it is the field itself.
  * For complex foreign keys collect field in one contiguous buffer.
  * Try to place resulting key in @a *buffer, that must be a buffer of size
  * COMPLEX_KEY_BUFFER_SIZE. If there's not enough space - allocate needed
  * using xmalloc - that pointer is returned via @a buffer. Thus if the pointer
  * is changed - a user of that function must free() the buffer after usage.
- * Return pointer to ready-to-use key in any case.
+ * In any case, the pointer to ready-to-use key is returned in @a key.
+ * Return codes:
+ *  1 on success (all parts of the key are non-null);
```

Comment:
We usually try to return 0 on success and -1 on error. I think it's better to return 0 and set *key to NULL if all parts are null.

Reply:
Done